test/TestCases/Expressions/Add.cpp

```diff
@@ -54,4 +54,36 @@ TEMPLATE_PRODUCT_TEST_CASE ("Addition in place", "[add]", (PlatformVectorOps, VC
 
     REQUIRE_THAT (sum1, vctr::EqualsTransformedBy<addition> (srcA, srcB).withEpsilon());
     REQUIRE_THAT (sum2, vctr::EqualsTransformedBy<addition> (srcA, c).withEpsilon());
+
+    // Operator += implements special accelerated handling for the case of multiply-accumulate operations
+    auto mac1 = srcA;
+    auto mac2 = srcA;
+    //auto mac3 = srcA;
+
+    mac1 += srcB * srcC;
+    mac2 += srcB * c;
+    //mac3 += vctr::multiplyByConstant<42> << srcB;
```

Comment:
This one did not build due to some compiler errors out of the scope of this task. As it's quite exotic I'll rather add a task to the backlog to fix that. Issue link is coming soon... ;)